pkg/compose/build.go

```diff
@@ -172,37 +161,32 @@ func (s *composeService) ensureImagesExists(ctx context.Context, project *types.
 	return nil
 }
 
-func (s *composeService) getBuildOptions(project *types.Project, images map[string]string) (map[string]build.Options, error) {
-	opts := map[string]build.Options{}
-	for _, service := range project.Services {
-		if service.Image == "" && service.Build == nil {
-			return nil, fmt.Errorf("invalid service %q. Must specify either image or build", service.Name)
+func (s *composeService) prepareProjectForBuild(project *types.Project, images map[string]string) error {
+	err := api.BuildOptions{}.Apply(project)
```

Comment:
Do we need to actually pass the `api.BuildOptions` down?

In `Apply` we use `Pull` / `NoCache`, which I assume are for the global CLI `--pull` & `--no-cache` flags respectively

Reply:
we only expose those flags for `build` command. `up --build` only applies options set by compose file